src/UserService.Business/Commands/Education/RemoveEducationCommand.cs

```diff
@@ -1,56 +1,113 @@
-        private readonly IEducationRepository _educationRepository;
-
-        public RemoveEducationCommand(
-            IAccessValidator accessValidator,
-            IHttpContextAccessor httpContextAccessor,
-            IUserRepository userRepository,
-            IEducationRepository educationRepository)
-        {
-            _accessValidator = accessValidator;
-            _httpContextAccessor = httpContextAccessor;
-            _userRepository = userRepository;
-            _educationRepository = educationRepository;
-        }
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _userRepository = userRepository;
+      _imageRepository = imageRepository;
+      _educationRepository = educationRepository;
+      _logger = logger;
+      _removeImagesRequest = removeImagesRequest;
+    }
+
+    private async Task<bool> RemoveImages(List<Guid> imagesToRemove, List<string> errors)
+    {
+      string errorMsg = "Can not remove education images. Reason: {errors}";
+
+      try
+      {
+        Response<IOperationResult<bool>> response = await
+          _removeImagesRequest.GetResponse<IOperationResult<bool>>(
+          IRemoveImagesRequest.CreateObj(imagesToRemove, ImageSource.User));
+
+        IOperationResult<bool> responsedMsg = response.Message;
 
-        public OperationResultResponse<bool> Execute(Guid educationId)
+        if (responsedMsg.IsSuccess)
         {
-            var senderId = _httpContextAccessor.HttpContext.GetUserId();
-            var dbUser = _userRepository.Get(senderId);
-            DbUserEducation userEducation = _educationRepository.Get(educationId);
-
-            if (!(dbUser.IsAdmin ||
-                  _accessValidator.HasRights(Rights.AddEditRemoveUsers))
-                  && senderId != userEducation.UserId)
-            {
-                throw new ForbiddenException("Not enough rights.");
-            }
-
-            bool result = _educationRepository.Remove(userEducation);
-
-            return new OperationResultResponse<bool>
-            {
-                Status = OperationResultStatusType.FullSuccess,
-                Body = result
-            };
+          return responsedMsg.Body;
         }
+
+        _logger.LogWarning(errorMsg, string.Join(',', responsedMsg.Errors));
+      }
+      catch (Exception e)
+      {
+        _logger.LogError(e, errorMsg);
+      }
+
+      errors.Add("Cannot remove education images");
+      return false;
+    }
+
+    public async Task<OperationResultResponse<bool>> Execute(Guid educationId)
+    {
+      OperationResultResponse<bool> result = new();
+      List<string> resultErrors = result.Errors;
+
+      Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+      DbUser sender = _userRepository.Get(senderId);
+      DbUserEducation userEducation = _educationRepository.Get(educationId);
+
+      if (!(sender.IsAdmin || _accessValidator.HasRights(Rights.AddEditRemoveUsers)) 
```

Comment:
set last kernel version and remove admin check

Reply:
++